### PR TITLE
Fix Neko Bytecode parser

### PR DESCRIPTION
**Pull request by Vtec234**
The parser was basically broken. This fixes it and brings it closer to what's done in Neko's own parser ([link](https://github.com/HaxeFoundation/neko/blob/01172c64f92f3d0bc8503ec76f251d46c0af4484/src/neko/Bytecode.nml))

I made the VM compile with these changes, but it's still mostly incomplete.

### CODE REVIEW

File: format/neko/VM.hx [hunk 1]
```diff
@@ -1,625 +1,586 @@
...
+					error(pc, "Invalid array access");
+				}
+			case OAccIndex(v):
+				acc = accIndex(pc, acc, v);
+			case OAccIndex0:
+				acc = accIndex(pc, acc, 0);
+			case OAccIndex1:
+				acc = accIndex(pc, acc, 1);
+			case OAccBuiltin(v):
+				error(pc, "TODO");
```
Comment (Aurel300) [inline format/neko/VM.hx]: This is breaking something that worked before, though.

File: format/neko/Reader.hx [hunk 2]
```diff
@@ -1,163 +1,310 @@
...
+				GlobalDebug(readDebugInfos());
+			case 6:
+				GlobalVersion(i.readByte());
+			default:
+				error();
+			}
+
+		// fields
+		var fields = alloc(nfields);
+		var fieldHashes : Map<Int, String> = [];
```
Comment (Aurel300) [inline format/neko/Reader.hx]: For Haxe 3 compatibility, this should be `var fieldHashes = new Map<Int, String>();`

**Comment by ncannasse** [comment]
Line endings were changed, resulting in all lines being part of the diff, please fix so I can review properly.

**Comment by ncannasse** [comment]
Is the PR complete?

**Comment by Aurel300** [comment]
@ncannasse The updated VM is still missing implementations of many ops (e.g. arithmetic), but the parser is more complete/less broken than before. I don't think Vtec wants to add the arithmetic instructions.

The CI failure is unrelated (problem in amf3).

**Pull request merged**